### PR TITLE
refactor: remove dead code from BucketOverlayForm

### LINKED ISSUE #2078: Chore: remove dead code from BucketOverlayForm
This code here seems to be of no use now:
https://github.com/influxdata/ui/blob/master/src/buckets/components/BucketOverlayForm.tsx#L161-L178

Looks like we had a text box input and hence we needed a validator to make sure the provided retention period was greater than `MIN_RETENTION_SECONDS`. https://github.com/influxdata/ui/commit/5fd36d285ed059e283533aa42cb420a084689392

We now use a dropdown and don't need to validate, there is no custom input.

### PR DESCRIPTION
**Pull request by ChitlangeSahas**
Closes #2078

Just removing some unnecessary code I came across when trying to remove `moment` from our codebase. (#1844)

### CODE REVIEW

File: src/buckets/components/BucketOverlayForm.tsx [hunk 1]
```diff
@@ -1,13 +1,11 @@
 // Libraries
 import React, {PureComponent, ChangeEvent, FormEvent} from 'react'
-import moment from 'moment'
 
 // Components
 import {Form, Input, Button, Grid} from '@influxdata/clockface'
 import Retention from 'src/buckets/components/Retention'
 
 // Constants
-import {MIN_RETENTION_SECONDS} from 'src/buckets/constants'
```
Comment (hoorayimhelping) [inline src/buckets/components/BucketOverlayForm.tsx:10]: also, you can delete this from `src/buckets/constants/index.ts` - i just did a quick global search and it's only used here.

**Review by hoorayimhelping — APPROVED** [review]
good work! red diffs >>>>> green diifs.

do a global search for `MIN_RETENTION_SECONDS` - it's not used anywhere else, so ✂️ it

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 57 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 100 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=2079)

File: src/buckets/components/BucketOverlayForm.tsx (hunk 1 above)
Comment (ChitlangeSahas) [reply src/buckets/components/BucketOverlayForm.tsx:10]: Will do yes!!

Comment (ChitlangeSahas) [reply src/buckets/components/BucketOverlayForm.tsx:10]: ✅

**Pull request merged**